src/transformers/models/opt/modeling_opt.py

```diff
@@ -359,9 +362,106 @@ def forward(
         return attn_output, attn_weights_reshaped, past_key_value
 
 
+class OPTSdpaAttention(OPTAttention):
```

Comment:
I'd be more in favor of using `# Copied from ...` here tbh (as long as it doesn't break anything).

Reply:
I didn't use copy because the other attn implementations are not really copy. the sdpa is also not a perfect copy as well

Reply:
Ah ok, makes sense. If there was an option to disable copying for a few lines, it'd favor that more. For now, I think it's ok to leave it like this.